pkg/image/builder.go

```diff
@@ -80,23 +82,45 @@ func RewriteImages(srcRegistry, destRegistry registrytypes.RegistryOptions, appS
 
 func GetPrivateImages(baseDir string, kotsKindsImages []string, checkedImages map[string]types.ImageInfo, allPrivate bool, dockerHubRegistry registrytypes.RegistryOptions, parentHelmChartPath string, useHelmInstall map[string]bool) ([]string, []k8sdoc.K8sDoc, error) {
 	uniqueImages := make(map[string]bool)
-
 	objectsWithImages := make([]k8sdoc.K8sDoc, 0) // all objects where images are referenced from
 
-	for _, image := range kotsKindsImages {
+	var mtx sync.Mutex
+	const concurrencyLimit = 10
+	g, _ := errgroup.WithContext(context.Background())
+	g.SetLimit(concurrencyLimit)
+
+	isPrivateImage := func(image string) (bool, error) {
 		if allPrivate {
-			checkedImages[image] = types.ImageInfo{
-				IsPrivate: true,
-			}
-		} else if _, ok := checkedImages[image]; !ok {
-			isPrivate, err := IsPrivateImage(image, dockerHubRegistry)
-			if err != nil {
-				return nil, nil, errors.Wrapf(err, "failed to check if kotskinds image %s is private", image)
-			}
-			checkedImages[image] = types.ImageInfo{
-				IsPrivate: isPrivate,
-			}
+			return true, nil
+		}
+
+		mtx.Lock()
+		checkedImage, ok := checkedImages[image]
+		mtx.Unlock()
+		if ok {
+			return checkedImage.IsPrivate, nil
+		}
+
+		p, err := IsPrivateImage(image, dockerHubRegistry)
+		if err != nil {
+			return false, err
 		}
+		return p, nil
+	}
+
+	for _, image := range kotsKindsImages {
+		func(image string) {
```

Comment:
unfortunately, the `golang.org/x/sync/errgroup` module doesn't allow passing/capturing variables. so I had to do it this way. capturing the variable is necessary so that its value doesn't change when the goroutine is triggered.

Here's a small go playground demonstration for what I mean: https://go.dev/play/p/Iepe6zjGZmK